In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("data/raw_credit_data.csv")

In [3]:
df.shape

(32579, 12)

In [4]:
df.info

<bound method DataFrame.info of        person_age  person_income person_home_ownership  person_emp_length  \
0              21           9600                   OWN                5.0   
1              25           9600              MORTGAGE                1.0   
2              23          65500                  RENT                4.0   
3              24          54400                  RENT                8.0   
4              21           9900                   OWN                2.0   
...           ...            ...                   ...                ...   
32574          57          53000              MORTGAGE                1.0   
32575          54         120000              MORTGAGE                4.0   
32576          65          76000                  RENT                3.0   
32577          56         150000              MORTGAGE                5.0   
32578          66          42000                  RENT                2.0   

           loan_intent loan_grade  loan_amn

In [5]:
df.describe()

,person_age,person_income,person_emp_length,loan_amnt,loan_int_rate,loan_status,loan_percent_income,cb_person_cred_hist_length
count,32579.000000,3.257900e+04,31684.000000,32579.000000,29463.000000,32579.000000,32579.000000,32579.000000
mean,27.734983,6.607120e+04,4.782224,9588.271586,11.011677,0.218147,0.170193,5.804352
std,6.348084,6.198108e+04,4.034888,6320.449757,3.240333,0.412994,0.106759,4.055084
min,20.000000,4.000000e+03,0.000000,500.000000,5.420000,0.000000,0.000000,2.000000
25%,23.000000,3.850000e+04,2.000000,5000.000000,7.900000,0.000000,0.090000,3.000000
50%,26.000000,5.500000e+04,4.000000,8000.000000,10.990000,0.000000,0.150000,4.000000
75%,30.000000,7.920000e+04,7.000000,12200.000000,13.470000,0.000000,0.230000,8.000000
max,144.000000,6.000000e+06,41.000000,35000.000000,23.220000,1.000000,0.830000,30.000000


In [6]:
df.isnull().sum()

person_age                       0
person_income                    0
person_home_ownership            0
person_emp_length              895
loan_intent                      0
loan_grade                       0
loan_amnt                        0
loan_int_rate                 3116
loan_status                      0
loan_percent_income              0
cb_person_default_on_file        0
cb_person_cred_hist_length       0
dtype: int64

In [7]:
missing_percent = df.isnull().mean() * 100
missing_percent.sort_values(ascending=False)

loan_int_rate                 9.564443
person_emp_length             2.747168
person_age                    0.000000
person_income                 0.000000
person_home_ownership         0.000000
loan_intent                   0.000000
loan_grade                    0.000000
loan_amnt                     0.000000
loan_status                   0.000000
loan_percent_income           0.000000
cb_person_default_on_file     0.000000
cb_person_cred_hist_length    0.000000
dtype: float64

## <font color='red'>Null Values</font>

Interest rate (loan_int_rate): 9.56% (3116)
<br>
Employment length in years (person_emp_length): 2.75% (895)

#### Employment length in years (person_emp_length)

Empty fields are set to 0

In [8]:
df["emp_length_missing"] = df["person_emp_length"].isnull().astype(int)
df["person_emp_length"] = df["person_emp_length"].fillna(df["person_emp_length"].median())

In [9]:
df["person_emp_length"].isnull().sum()

np.int64(0)

#### Interesst rate (loan_int_rate)

In [10]:
df[df["loan_int_rate"].isnull()]["loan_status"].value_counts()

loan_status
0    2472
1     644
Name: count, dtype: int64

#### Loan

Loan interest rate: "null" are removed

In [11]:
print(df["loan_int_rate"].isnull().sum())

# Drop rows with missing interest rate
df = df.dropna(subset=["loan_int_rate"])

# Verify
print(df["loan_int_rate"].isnull().sum())
print(df.shape)

3116
0
(29463, 13)


In [12]:
df["loan_int_rate"].isnull().sum()

np.int64(0)

In [13]:
df.shape

(29463, 13)

#### Age

People older than 85 are removed

In [14]:
df[df["person_age"] >= 85].shape

(4, 13)

In [15]:
df = df[df["person_age"] <= 85].copy()

In [16]:
df[df["person_age"] > 85].shape

(0, 13)

In [17]:
df.shape

(29459, 13)

#### Working

People how have been working bevor they turned 18

In [18]:
df.drop(columns=["emp_length_missing"], inplace=True)
# df.to_csv("data/cleaned_credit_data.csv", index=False)

In [19]:
df.shape

(29459, 12)